qa/L0_backend_python/io/io_test.py

```diff
@@ -28,45 +28,74 @@
-                for model_1_in_gpu in [True, False]:
-                    for model_2_in_gpu in [True, False]:
-                        for model_3_in_gpu in [True, False]:
-                            gpu_output = np.asarray([
-                                model_1_in_gpu, model_2_in_gpu, model_3_in_gpu
-                            ],
-                                                    dtype=bool)
-                            inputs = [
-                                httpclient.InferInput(
-                                    "INPUT0", input0.shape,
-                                    np_to_triton_dtype(input0.dtype)),
-                                httpclient.InferInput(
-                                    "GPU_OUTPUT", gpu_output.shape,
-                                    np_to_triton_dtype(gpu_output.dtype))
-                            ]
-                            inputs[0].set_data_from_numpy(input0)
-                            inputs[1].set_data_from_numpy(gpu_output)
-                            result = client.infer(model_name, inputs)
-                            output0 = result.as_numpy('OUTPUT0')
-                            self.assertIsNotNone(output0)
-                            self.assertTrue(np.all(output0 == input0))
+        user_data = UserData()
+        with grpcclient.InferenceServerClient("localhost:8001") as client:
+            input0 = np.random.random([1000]).astype(np.float32)
+            client.start_stream(callback=partial(callback, user_data))
+            for model_1_in_gpu in [True, False]:
+                for model_2_in_gpu in [True, False]:
+                    for model_3_in_gpu in [True, False]:
+                        gpu_output = np.asarray(
+                            [model_1_in_gpu, model_2_in_gpu, model_3_in_gpu],
+                            dtype=bool)
+                        inputs = [
+                            grpcclient.InferInput(
+                                "INPUT0", input0.shape,
+                                np_to_triton_dtype(input0.dtype)),
+                            grpcclient.InferInput(
+                                "GPU_OUTPUT", gpu_output.shape,
+                                np_to_triton_dtype(gpu_output.dtype))
+                        ]
+                        inputs[0].set_data_from_numpy(input0)
+                        inputs[1].set_data_from_numpy(gpu_output)
+                        client.async_stream_infer(model_name=model_name,
+                                                  inputs=inputs)
+                        result = user_data._completed_requests.get()
+                        output0 = result.as_numpy('OUTPUT0')
+                        self.assertIsNotNone(output0)
+                        self.assertTrue(np.all(output0 == input0))
+
+    def test_ensemble_io(self):
+        # Only run the shared memory leak detection with the default trial
+        if TRIAL == 'default':
```

Comment:
$TRIAL == 'default'

Reply:
This is Python script with TRIAL defined above.